### DISCUSSION TITLE
Beginning user of RABIES with Singularity installation

### DISCUSSION THREAD

**Discussion by geowk** [CoBrALab/RABIES#286] [General]
Hello,

Great tool to use.

I had a few questions for a beginner:

1. I've installed using the singularity method on a linux computer: With this method, how  does one run a simple command such as rabies -help and the output of the command writes to screen? In the instructions, it gives a few examples of "rabies --help". That doesn't seem to work if you installed using a singularity container.
2. The MultiProcess option:  Does this mean running multiple subjects in parallel (e.g. 5 subjects in input_BIDS file, each processed using 5 cpu cores, 1 subject per cpu core? Can it used multiple cores with 1 subject as input?
3. Orientation of the data: Does one have to reorient the images to match the L/R conventions of the templates used (Sigma Rat Atlas) before running RABIES?

Thanks.

g.

**Comment by Gab-D-G** [comment]
Hi! 
1. When using singularity, you must recruit the rabies program using 'singularity run' and the rabies singularity file: `singularity run /path/to/rabies.sif --help`
2. This option will execute RABIES steps in parallel, matching the maximum number of cores available (unless you manually specify otherwise with --local_threads). This means running steps from different subjects in parallel, as well as different preprocessing steps for the same subject, if possible.
3. RABIES will re-orient all images to the RAS orientation, including the template files, if they are not already. So if the metadata for your images and template are consistent, it should not be an issue. You should double check though when RABIES is running (you can inspect whether the orientation is matching between the files displayed in the preprocess_QC_report).

Hope this helps. - Gabriel DG

**Comment by geowk** [comment]
Thank you Gabriel for your response. It was very helpful.

Some more questions for a beginner: From doing some background reading, RABIES is broken into 3 phases: Preprocessing, Confound Correction, and Analysis.

1. Does RABIES have the capability of running Confound Correction on preprocessed data from another software?
 For example, say we preprocessed the data with another software, but want to use the Confound Correction of RABIES. Could we simply use the preprocessed data (in BIDS and nifti format) and just run the RABIES Confound Correction?

2. What if we just want to run the Analysis portion of RABIES on a BIDS formatted dataset that wasn't preprocessed/cleaned by RABIES, but another software?

Thanks.

g.

**Comment by Gab-D-G** [comment]
Hi, 
1. Unfortunately, RABIES is well adapted for this purpose. The confound correction and analysis workflow require a set of outputs from the preprocessing stage. It would be possible to use outputs from another software by creating RABIES-like preprocessing outputs, reproducing the datasink folders and appropriate file naming. Then, it would be possible to use the `--read_datasink` option from the confound correction stage and carry on from there (if all the expected outputs are found). With that said, this is not what RABIES is currently designed to do.
2. It would be the same issue as in 1. RABIES is not primarily an analysis software; the analysis options are meant to be an extension of the resources provided for image processing. There are many alternatives to carry out similar basic connectivity measures without using RABIES if your goal is not preprocessing and confound correction.

Gabe DG

**Comment by geowk** [comment]
Thanks again, Gabe.

Can RABIES be used for both rat and mouse data? We plan to use the Sigma Rat Atlas for the rat brain as the template and the Allen Brain for mouse data as the template. Are these available within RABIES by default or do we have to add the files with their directory path on our host machine using the template option in preprocessing?

g.

**Comment by Gab-D-G** [comment]
RABIES works well for both mice and rats. We are planning to add the SIGMA template (see issue #255 ), but it is not currently available by defaults. You must specify it manually during preprocessing. When using singularity, you will need to set the paths leading to your SIGMA template files with -B, similarly to how you set the path to your input/output directory (see https://rabies.readthedocs.io/en/latest/running_the_software.html for more details.)

**Comment by geowk** [comment]
Thanks.

I tried to run a quick RABIES command with rat dataset and I think I'm missing something when it comes to assigning the anat_template for the Sigma Rat Atlas on my computer and the -B setup for the template folder.

The error generated says the file doesn't exist, but it exist in the appropriate directory on my local machine.

Here is the code

```

singularity run -B /data/example/rawdata/derivatives/sigma/:/template/   \
-B sub-901:/input_BIDS:ro -B preprocess_outputs:/preprocess_outputs/ \
/data/example/rawdata/derivatives/preproc/rabies/sing/rabies.sif -p MultiProc --local_threads 15 \
preprocess /input_BIDS/ /preprocess_outputs  \
--anat_template /data/example/rawdata/derivatives/sigma/SIGMA_InVivo_Brain_Template.nii \
--brain_mask /data/example/rawdata/derivatives/sigma/SIGMA_InVivo_Brain_Mask.nii \
--WM_mask /data/example/rawdata/derivatives/sigma/SIGMA_InVivo_WM.nii \
--CSF_mask /data/example/rawdata/derivatives/sigma/SIGMA_InVivo_CSF.nii \
--vascular_mask /data/example/rawdata/derivatives/sigma/SIGMA_InVivo_CSF.nii \
--labels /data/example/rawdata/derivatives/sigma/SIGMA_InVivo_Anatomical_Brain_Atlas.nii --TR 1.0 \
--commonspace_resampling 0.4x0.4x0.4


230511-19:44:34,236 nipype.workflow INFO:
	 Running RABIES - version: 0.4.8
230511-19:44:34,237 nipype.workflow INFO:
	 CLI INPUTS: 
-> rabies_stage = preprocess 
-> plugin = MultiProc 
-> local_threads = 15 
-> scale_min_memory = 1.0 
-> min_proc = 1 
-> verbose = 1 
-> bids_dir = /input_BIDS 
-> output_dir = /preprocess_outputs 
-> bold_only = False 
-> anat_autobox = False 
-> bold_autobox = False 
-> apply_despiking = False 
-> HMC_option = intraSubjectBOLD 
-> apply_slice_mc = False 
-> detect_dummy = False 
-> data_type = float32 
-> anat_inho_cor = {'method': 'SyN', 'otsu_thresh': '2', 'multiotsu': False} 
-> anat_robust_inho_cor = {'apply': False, 'masking': False, 'brain_extraction': False, 'template_registration': 'SyN'} 
-> bold_inho_cor = {'method': 'Rigid', 'otsu_thresh': '2', 'multiotsu': False} 
-> bold_robust_inho_cor = {'apply': False, 'masking': False, 'brain_extraction': False, 'template_registration': 'SyN'} 
-> commonspace_reg = {'masking': False, 'brain_extraction': False, 'template_registration': 'SyN', 'fast_commonspace': False} 
-> bold2anat_coreg = {'masking': False, 'brain_extraction': False, 'registration': 'SyN'} 
-> nativespace_resampling = inputs_defined 
-> commonspace_resampling = 0.4x0.4x0.4 
-> anatomical_resampling = inputs_defined 
-> apply_STC = False 
-> TR = 1.0 
-> tpattern = alt-z 
-> stc_axis = Y 
-> anat_template = /data/example/rawdata/derivatives/sigma/SIGMA_InVivo_Brain_Template.nii 
-> brain_mask = /data/example/rawdata/derivatives/sigma/SIGMA_InVivo_Brain_Mask.nii 
-> WM_mask = /data/example/rawdata/derivatives/sigma/SIGMA_InVivo_WM.nii 
-> CSF_mask = /data/example/rawdata/derivatives/sigma/SIGMA_InVivo_CSF.nii 
-> vascular_mask = /data/example/rawdata/derivatives/sigma/SIGMA_InVivo_CSF.nii 
-> labels = /data/example/rawdata/derivatives/sigma/SIGMA_InVivo_Anatomical_Brain_Atlas.nii 



230511-19:44:34,237 nipype.workflow INFO:
	 INPUT BIDS DATASET:  
input_BIDS/ 
    ses-01/ 
        anat/ 
            sub-1001_ses-01_T2w.nii 
            sub-1001_ses-01_T2w.json 
        func/ 
            sub-1001_ses-01_task-rest_bold.nii 
            sub-1001_ses-01_task-rest_bold.json 

Traceback (most recent call last):
  File "/home/rabies/miniforge/bin/rabies", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/rabies/RABIES/scripts/rabies", line 3, in <module>
    execute_workflow()
  File "/home/rabies/RABIES/rabies/run_main.py", line 44, in execute_workflow
    workflow = preprocess(opts, log)
  File "/home/rabies/RABIES/rabies/run_main.py", line 178, in preprocess
    raise ValueError(f"--anat_template file {opts.anat_template} doesn't exists.")
ValueError: --anat_template file /data/example/rawdata/derivatives/sigma/SIGMA_InVivo_Brain_Template.nii doesn't exists.


```

**Reply by Gab-D-G** [reply]
You have set the path to your template folder as /template. You must begin the path to your file from there when calling RABIES. It should be for instance --anat_template /template/SIGMA_InVivo_Brain_Template.nii , since /template leads to /data/example/rawdata/derivatives/sigma.

**Comment by geowk** [comment]
Oh, okay. Nice simple fix. It worked. Thank you, Gabe.

g.
